lib/FusionInventory/Agent/Task/Inventory/Win32/Softwares.pm

```diff
@@ -316,4 +339,55 @@ sub _processMSIE {
 
 }
 
+# List of SQL Instances
+sub _getSqlEdition {
+    my (%params) = @_;
+
+    my $softwareName = $params{softwarename};
+    my $softwareVersion = $params{softwareversion};
+
+    # Registry access for SQL Instances
+    my $sqlinstancesList = getRegistryKey(
+        path    => "HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Microsoft SQL Server/Instance Names/SQL"
+    );
+    return unless $sqlinstancesList;
+
+    # List of SQL Instances
+    foreach my $sqlinstance (keys %$sqlinstancesList) {
+        my $sqlinstanceName = $sqlinstance;
```

Comment:
You do not need to set a new value here, better redefine foreach loop with `foreach my $sqlinstanceName ...`